python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -634,10 +634,46 @@ std::string getASM(const std::string &name, MlirModule module,
   auto cloned = unwrap(module).clone();
   auto context = cloned.getContext();
 
+  // Get additional debug values
+  auto disableMLIRthreading = getEnvBool("CUDAQ_MLIR_DISABLE_THREADING", false);
+  auto enablePrintMLIREachPass =
+      getEnvBool("CUDAQ_MLIR_PRINT_EACH_PASS", false);
+
   PassManager pm(context);
   pm.addPass(cudaq::opt::createLambdaLiftingPass());
+  // Run most of the passes from hardware pipelines.
+
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
+  pm.addNestedPass<func::FuncOp>(cudaq::opt::createClassicalMemToReg());
+  pm.addPass(cudaq::opt::createLoopUnroll());
+  pm.addPass(createCanonicalizerPass());
+  pm.addNestedPass<func::FuncOp>(cudaq::opt::createLiftArrayAlloc());
+  pm.addPass(cudaq::opt::createGlobalizeArrayValues());
+  pm.addPass(cudaq::opt::createStatePreparation());
+  pm.addNestedPass<func::FuncOp>(cudaq::opt::createGetConcreteMatrix());
+  pm.addPass(cudaq::opt::createUnitarySynthesis());
+  pm.addPass(cudaq::opt::createApplyOpSpecializationPass());
+  cudaq::opt::addAggressiveEarlyInlining(pm);
+  pm.addPass(createSymbolDCEPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
+  pm.addNestedPass<func::FuncOp>(
+      cudaq::opt::createMultiControlDecompositionPass());
+  pm.addPass(cudaq::opt::createDecompositionPass(
+      {.enabledPatterns = {"SToR1", "TToR1", "R1ToU3", "U3ToRotations",
```

Comment:
`SToR1` and `TToR1` may not be required on all OpenQASM2.0 targets, only those which don't support `sdg` and `tdg` out of the box. Similarly, we should remove `R1toU3` and `U3ToRotations` from generic workflow.

Reply:
we need to rethink python's cudaq::translate implementation so it can use the target's pipeline instead. Will leave to the later time.

Reply:
This can wait but if I understood correctly, these changes to the `getASM` function do not affect the `braket` target, right?
And this function will be invoked only when users use the API `cudaq.translate(<kernel>, format="openqasm2")`, correct?
If yes to both, only the generic passes should be enabled here for translation to OpenQASM2.0, and `"SToR1", "TToR1", "R1ToU3", "U3ToRotations"` should be removed from here.

Reply:
Do we need to open an issue for this comment thread?